### PR TITLE
add namespace for envoyfilter.

### LINKED ISSUE #18878: Operator controller gets reconciling error: Namespace parameter required.
**Bug description**

With the latest master, the operator controller always complain the error `Namespace parameter required.` while updateing the resources, result in incorrect ICP status:
```
status:
  status:
    Base:
      status: 2
      statusString: HEALTHY
    Citadel:
      status: 2
      statusString: HEALTHY
    Galley:
      status: 2
      statusString: HEALTHY
    Grafana:
      status: 2
      statusString: HEALTHY
    IngressGateway:
      status: 2
      statusString: HEALTHY
    Injector:
      status: 2
      statusString: HEALTHY
    Pilot:
      error: Namespace parameter required.
      status: 3
      statusString: ERROR
    Policy:
      status: 2
      statusString: HEALTHY
    Telemetry:
      status: 2
      statusString: HEALTHY
    Tracing:
      status: 2
      statusString: HEALTHY
```

**Expected behavior**

**Steps to reproduce the bug**

**Version (include the output of `istioctl version --remote` and `kubectl version`)**

**How was Istio installed?**

**Environment where bug was observed (cloud vendor, OS, etc)**

### PR DESCRIPTION
**Pull request by morvencao**
resolve: https://github.com/istio/istio/issues/18878

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]
this technically should be the root namespace which may not be the same as release. I think global.rootNamespaceor something

File: istio-control/istio-discovery/metadata-exchange-v2.yaml [hunk 1]
```diff
@@ -2,6 +2,7 @@ apiVersion: networking.istio.io/v1alpha3
 kind: EnvoyFilter
 metadata:
   name: metadata-exchange
+  namespace: {{ .Release.Namespace }}
```
Comment (mandarjog) [inline istio-control/istio-discovery/metadata-exchange-v2.yaml]: Is this the config namespace? We want these to apply to all namespaces.

**Comment by morvencao** [comment]
Looks like the imported files will not be send to helm render, so the added `namespace: {{ .Release.Namespace }}` will not be rendered correctly.

File: istio-control/istio-discovery/metadata-exchange-v2.yaml (hunk 1 above)
Comment (morvencao) [reply istio-control/istio-discovery/metadata-exchange-v2.yaml]: should it be in `global.configNamespace`?

**Comment by richardwxn** [comment]
but is EnvoyFilter scoped to namespace?

**Comment by morvencao** [comment]
Yes, envoyfiler is namespace scoped resource.

I have added `global.configNamespace` to the two config files.

**Comment by howardjohn** [comment]
configNamespace is not right, that is where galley lives. Our whole namespace setup isn't great, but we need this code from istio/istio in the mesh config -- right now it is missing:

https://github.com/istio/istio/blob/2b971c71aa807f777219c1bf64ed6932770d6f06/install/kubernetes/helm/istio/templates/configmap.yaml#L157-L161

**Comment by morvencao** [comment]
@howardjohn Would you take another look about the change set? I have updated the PR.

**Review by howardjohn — APPROVED** [review]

**Comment by howardjohn** [comment]
@morvencao I approved this, not sure if you want the hold still

**Pull request merged**